modules/commons/src/test/java/org/apache/synapse/commons/json/JsonValueTest.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class JsonValueTest extends TestCase {
+    private static Log logger = LogFactory.getLog(JsonValueTest.class.getName());
+
+    private static final String stringVal = "\"Hello world!\"";
+    private static final String nullVal = "null";
+    private static final String trueVal = "true";
+    private static final String falseVal = "false";
+    private static final String numberVal1 = "12.5";
+    private static final String numberVal2 = "-12.5";
+    private static final String numberVal3 = "123";
+    private static final String failTestVal1 = "123abc";
+    private static final String failTestVal2 = "task";
+    private static final String failTestVal3 = "fun";
+    private static final String failTestVal4 = "abc";
+    private static final String failTestVal5 = "-abc";
+    private static final String failTestVal6 = "\"Hello world!";
+
+    public void testCase1() {
+        runTest(stringVal);
+    }
+
+    public void testCase2() {
+        runTest(nullVal);
+    }
+
+    public void testCase3() {
+        runTest(trueVal);
+    }
+
+    public void testCase4() {
+        runTest(falseVal);
+    }
+
+    public void testCase5() {
+        runTest(numberVal1);
+    }
+
+    public void testCase6() {
+        runTest(numberVal2);
+    }
+
+    public void testCase7() {
+        runTest(numberVal3);
+    }
+
+    public void runTest(String value) {
+        try {
+            InputStream inputStream = Util.newInputStream(value.getBytes());
+            MessageContext messageContext = Util.newMessageContext();
+            OMElement element = JsonUtil.getNewJsonPayload(messageContext, inputStream, true, true);
+            OutputStream out = Util.newOutputStream();
+            JsonUtil.writeAsJson(element, out);
+            assertTrue(value.equals(out.toString()));
+        } catch (Exception ex) {
+            logger.error(ex.getMessage(), ex);
+            assertTrue(false);
+        }
+    }
+
+    public void testCase8() {
+        runFailTest(failTestVal1);
+    }
+
+    public void testCase9() {
+        runFailTest(failTestVal2);
+    }
+
+    public void testCase10() {
+        runFailTest(failTestVal3);
+    }
+
+    public void testCase11() {
+        runFailTest(failTestVal4);
+    }
+
+    public void testCase12() {
+        runFailTest(failTestVal5);
+    }
+
+    public void testCase13() {
+        runFailTest(failTestVal6);
+    }
+
+    public void runFailTest(String value) {
+        try {
+            InputStream inputStream = Util.newInputStream(value.getBytes());
+            MessageContext messageContext = Util.newMessageContext();
+            JsonUtil.getNewJsonPayload(messageContext, inputStream, true, true);
+            assert false;
```

Comment:
Shall we use the method, fail() since it's more explicit?

Reply:
Fixed in #759